man/kernel-install.xml

```diff
@@ -111,9 +111,8 @@
 
             <listitem><para><filename>90-uki-copy.install</filename> copies a file
             <filename>uki.efi</filename> from <varname>$KERNEL_INSTALL_STAGING_AREA</varname> or if it does
-            not exist the <replaceable>KERNEL-IMAGE</replaceable> argument, iff it has a
-            <literal>.efi</literal> extension, to
-            <filename>$BOOT/EFI/Linux/<replaceable>ENTRY-TOKEN</replaceable>-<replaceable>KERNEL-VERSION</replaceable>.efi</filename>.</para>
+            not exist the <replaceable>KERNEL-IMAGE</replaceable> argument, if it is a unified kernel image, to
+            <filename>$ESP/EFI/Linux/<replaceable>ENTRY-TOKEN</replaceable>-<replaceable>KERNEL-VERSION</replaceable>.efi</filename>.</para>
```

Comment:
`$BOOT` is used here, because that's the language used in [BLS](https://uapi-group.org/specifications/specs/boot_loader_specification/#the-boot-partition-placeholder)

Reply:
They can be though. If I remember the spec correctly, UKis also don't need to be in the ESP, they can also be in `XBOOTLDR`.

Reply:
Yes, `$BOOT` and `$ESP` can be identical.  That is the easy case.  But what if they are not?  One advantage of UKIs is that they can also be booted without any bootloader, directly from the firmware, and you loose that if they are placed on a filesystem the firmware can't read.

Reply:
That is completely true and therefore the [spec says](https://uapi-group.org/specifications/specs/boot_loader_specification/#the-partitions):
> For systems where the firmware is able to read file systems directly, the ESP must — and the MBR boot and GPT XBOOTLDR partition should — be a file system readable by the firmware. For most systems this means VFAT (16 or 32 bit). Applications accessing both partitions should hence not assume that fancier file system features such as symlinks, hardlinks, access control or case sensitivity are supported.

This is also the reason why the ESP should not be mounted to `/boot/efi`, so that `/boot` and the ESP are not nested mounts and can be automounted.

Also, UKIs can be in `XBOOTLDR`, so writing `ESP` is too narrow.

Reply:
Reality often ignores the BLS recommendations.  Anaconda (the Fedora Installer) even throws errors in case I ask it to use vfat for /boot.

I can see that installing to `$BOOT` by default makes sense for consistency reasons, but I guess I want a config switch then so I have the option to force using `$ESP` instead.

Reply:
You already do, you can set `BOOT_ROOT` for the invocation of `kernel-install`.

Reply:
Same problem we have in the other discussion thread wrt. `layout=...`.  I can stick that into `install.conf`, but it'll be applied to both UKI and non-UKI kernels, which is not what I want.

Reply:
Maintainers of Anaconda refusing to drop the misguided and pointless check against vfat is hard to explain. There are no technical reasons for this. So we really shouldn't use Anaconda as justification for anything.

There are good reasons to allow installing to XBOOTLDR: the ESP might be too small. It won't work in all cases, but in some it certainly will, so it's a useful option to have.

Reply:
the whole concept of `$BOOT` exists only to be used as a *the* place to drop new kernels into, hence we should use it here.

not sure anaconda matters here, given it doesn't support XBOOTLDR anyway, does it? its boot partition does not use the xbootldr partition type, does it?

Reply:
FYI: @berrange made some progress on the anaconda front.

Both parted python bindings and blivet are green upstream:
https://bugzilla.redhat.com/show_bug.cgi?id=1075288

Anaconda itself is not (yet) ready:
https://bugzilla.redhat.com/show_bug.cgi?id=2160074

For now I'm doing `sfdisk` calls in `%post` in my test kickstart files.

I'm fine with using `$BOOT` by default but I need an override for use cases where I really need the UKI being placed on `$ESP`.